docs/content/getting-started/project-file-reference.mdx

```diff
@@ -0,0 +1,438 @@
+        <a href="/concepts/assets/software-defined-assets">software-defined assets</a>.
+        <br />
+        <br />
+        <strong>Note:</strong> As your project grows, we recommend organizing assets in sub-packages or sub-modules.
+        For example, you can put all analytics-related assets in a <code>my_dagster_project/assets/analytics/</code>
+        folder and use <PyObject object="load_assets_from_package_module" /> in the top-level definitions
+        to load them, rather than needing to manually add assets to the top-level definitions every time
+        you define one. Similarly, you can also use <PyObject object="load_assets_from_modules" /> to
+        load assets from single Python files. Refer to the <a href="/guides/dagster/example_project">Fully Featured Project guide</a> for more info and best practices.
+      </td>
+    </tr>
+
+  </tbody>
+</table>
+        </td>
+    </tr>
+    <tr>
+      <td>my_dagster_project_tests/</td>
+      <td>
+        A Python package that contains tests for <code>my_dagster_project</code>
+        .
+      </td>
+    </tr>
+    <tr>
+      <td>README.md</td>
+      <td>A description and starter guide for the Dagster project.</td>
+    </tr>
+    <tr>
+      <td>pyproject.toml</td>
+      <td>
+        A file that specifies package core metadata in a static, tool-agnostic
+        way.
+        <br />
+        <br />
+        This file includes a <code>tool.dagster</code> section which references the
+        Python package with your Dagster definitions defined and discoverable at
+        the top level. This allows you to use the
+        <code>dagster dev</code> command to load your Dagster code without any
+        parameters. Refer to the{" "}
+        <a href="/concepts/code-locations">Code locations documentation</a> to
+        learn more.
+        <br />
+        <br /> <strong>Note:</strong> <code>pyproject.toml</code> was introduced
+        in{" "}
+        <a href="https://peps.python.org/pep-0518/https://peps.python.org/pep-0518/">
+          PEP-518
+        </a>{" "}
+        and meant to replace <code>setup.py</code>, but we may still include a{" "}
+        <code>setup.py</code> for compatibility with tools that do not use this
+        spec.
+      </td>
+    </tr>
+    <tr>
+      <td>setup.py</td>
+      <td>
+        A build script with Python package dependencies for your new project as
+        a package.
+        <br /><br />
+        <strong>Note</strong>: If using Dagster Cloud, <code>dagster-cloud</code> as a dependency.
+      </td>
+    </tr>
+    <tr>
+      <td>setup.cfg</td>
+      <td>
+        An ini file that contains option defaults for <code>setup.py</code>{" "}
+        commands.
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+---
+
+## Configuration files
+
+<Note>
+  <strong>Not sure where to put these files?</strong> Unless otherwise noted,
+  all configuration files should be placed in the root of your Dagster project
```

Comment:
See comment, but I don't think this is generally true for most non-local dagster deployments. The recommendation would probably be to place these files (that control global dagster settings) alongside your infrastructure-as-code definitions. Though `dagster_cloud.yaml` would normally live in the project root.

Reply:
Not updated yet - see other comment: https://github.com/dagster-io/dagster/pull/13635#issuecomment-1522066388